### DISCUSSION TITLE
Support for VRC7 to OPL conversion?

### DISCUSSION THREAD

**Discussion by ccacook98** [digital-sound-antiques/vgm-conv#84] [Ideas]
I've had a look at the code, and it seems like it wouldn't be too difficult to make the YM2413-to-OPL converter also handle VRC7-to-OPL conversions. As it is, it works with existing VGM files containing VRC7 data, but converts them using the standard YM2413 voice table.

The underlying VGM library even has the VRC7 voice table already in it, but there's no way in the current version of `vgm-conv` to force it to use that table rather than the standard YM2413 one. I tried to add functionality to do this myself, but couldn't add the VRC7 as a new chip because the data about supported chips comes from the underlying VGM parser, which considers the VRC7 a YM2413 variant rather than an individual chip.

Another great idea might be to add voice tables for the other OPLL variants as well and allow the user to override the default one so that data logged from those chips can be converted as well.

Thanks for creating this awesome project!

**Comment by okaxaki** [comment]
Thank you for sugessions. 
I think underlying [vgm-parser](https://github.com/digital-sound-antiques/vgm-parser) should support VRC7. I will fix the vgm-parser first, then add VRC7 support to vgm-conv.
Note: As you may already know, VRC7 support is undocumented in VGM 1.71 spec., however in VGMPlay source, I've found the BIT31 of OPLL clock is treated as VRC7 flag.

I also think the voice table option is very flexible for OPLL variants. I'll give it some thought.

**Comment by okaxaki** [comment]
Update: VRC7 support has been implemented in [vgm-conv 0.14.0-beta.0](https://github.com/digital-sound-antiques/vgm-conv/blob/d0122b9436d237dd0bc302550fb0ca32b7c19594/CHANGELOG.md).
